examples/proof_aggregation/src/main.rs

```diff
@@ -0,0 +1,250 @@
+        <BtreeAggregation as PCDAggregation>::aggregate_and_prove_step(
+            // Input arity  = 2
+            &[aggregation_iv.clone(), aggregation_iv.clone()],
+            None,
+            &step_instance_1,
+            &step_proof_1,
+        );
+
+    let prove_duration = aggregation_base_case_timer.elapsed();
+    println!(
+        "Prove time (parallel aggregation base case): {:?}",
+        prove_duration
+    );
+
+    // The second instance to verify
+    let step_instance_2 = StepInstance::new(&compliance_instance_2, vk_2);
+    let step_proof_2 = StepProof(compliance_receipt_2);
+
+    // Second aggregation (base case)
+    let (aggregation_2, ag_proof_2) =
+        <BtreeAggregation as PCDAggregation>::aggregate_and_prove_step(
+            // Input arity  = 2
+            &[aggregation_iv.clone(), aggregation_iv],
+            None,
+            &step_instance_2,
+            &step_proof_2,
+        );
+
+    // The third instance to verify
+    let step_instance_3 = StepInstance::new(&compliance_instance_3, vk_3);
+    let step_proof_3 = StepProof(compliance_receipt_3);
+
+    let aggregation_recursive_case_timer = Instant::now();
+
+    // Third aggregation (recursive case)
+    let (aggregation_3, ag_proof_3) =
+        <BtreeAggregation as PCDAggregation>::aggregate_and_prove_step(
+            // Input arity  = 2
+            &[aggregation_1, aggregation_2],
+            Some(&[ag_proof_1, ag_proof_2]),
+            &step_instance_3,
+            &step_proof_3,
+        );
+
+    let recursive_prove_duration = aggregation_recursive_case_timer.elapsed();
+    println!(
+        "Prove time (parallel aggregation recursive case): {:?}",
+        recursive_prove_duration
+    );
+
+    /*
+     * Now, let's verify the third aggregation.
+     * Attests to correct validation of the three
+     * step instances.
+     * (In the high-level API, the verifier will derive the
+     * aggregated instance on their own.)
+     */
+    let verify_start_timer = Instant::now();
+
+    assert_eq!(
+        <BtreeAggregation as PCDAggregation>::verify_step(&aggregation_3, &ag_proof_3),
+        true
+    );
+
+    let verify_duration = verify_start_timer.elapsed();
+    println!("Verify parallel duration time: {:?}", verify_duration);
+}
+
+fn get_instances_programs_and_proofs_to_verify_mock() -> (
+    (ComplianceInstance, Receipt, [u32; 8]),
+    (ComplianceInstance, Receipt, [u32; 8]),
+    (ComplianceInstance, Receipt, [u32; 8]),
+) {
+    println!("Generating mock instances. This make take a while...");
+
+    let compliance_witness: ComplianceWitness<COMMITMENT_TREE_DEPTH> =
+        ComplianceWitness::<COMMITMENT_TREE_DEPTH>::default();
+
+    let env = ExecutorEnv::builder()
+        .write(&compliance_witness)
+        .unwrap()
+        .build()
+        .unwrap();
+
+    let prover = default_prover();
+
+    let compliance_receipt_1 = prover
+        .prove(env, compliance_methods::COMPLIANCE_GUEST_ELF)
```

Comment:
You can use aarm::constants::{COMPLIANCE_GUEST_ELF, COMPLIANCE_GUEST_ID} instead and remove the compliance-methods dependency.